### PR TITLE
Better handling of escaped quotes in textual item

### LINKED ISSUE #1551: `item` in python dict with a value `"\\"` selects multiple items
Input:
```
symbol_key_words = {
    "air": "each",
    "bat": "\\",
    "cap": "gust",
    "drum": "harp",
}
```
`take item air` selects the first item as expected.

`take item drum` selects the three last items in the example.

It works as expected with input:
```
symbol_key_words = {
    "air": "each",
    "bat": "fine",
    "cap": "gust",
    "drum": "harp",
}
```

### PR DESCRIPTION
**Pull request by AndreasArvidsson**
Fixes #1551

## Checklist

- [x] I have added [tests](https://www.cursorless.org/docs/contributing/test-case-recorder/)
- [ ] I have updated the [docs](https://github.com/cursorless-dev/cursorless/tree/main/docs) and [cheatsheet](https://github.com/cursorless-dev/cursorless/tree/main/cursorless-talon/src/cheatsheet)
- [ ] I have not broken the cheatsheet

### CODE REVIEW

File: packages/cursorless-engine/src/processTargets/modifiers/ItemStage/tokenizeRange.ts [hunk 1]
```diff
@@ -41,8 +41,8 @@ export function tokenizeRange(
   const lexemes = text
     // NB: Both the delimiters and the text between them are included because we
     // use a capture group in this split regex
-    .split(/([,(){}<>[\]"'`]|\\"|\\'|\\`)/g)
-    .filter((lexeme) => lexeme.length > 0);
+    .split(/([,(){}<>[\]"'`])|(?<!\\)(\\"|\\'|\\`)/g)
+    .filter(Boolean);
```
Comment (pokey) [inline packages/cursorless-engine/src/processTargets/modifiers/ItemStage/tokenizeRange.ts:45]: Why the switch to boolean?

**Review by pokey — APPROVED** [review]
LGTM with minor question

File: packages/cursorless-engine/src/processTargets/modifiers/ItemStage/tokenizeRange.ts (hunk 1 above)
Comment (AndreasArvidsson) [reply packages/cursorless-engine/src/processTargets/modifiers/ItemStage/tokenizeRange.ts:45]: Some of the items could be undefined. It's just a more terse way to do `.filter((lexeme) => lexeme != null && lexeme.length > 0);`

Comment (pokey) [reply packages/cursorless-engine/src/processTargets/modifiers/ItemStage/tokenizeRange.ts:45]: I generally lean toward verbose, and the conversion to boolean behaviour can be a bit unintuitive i believe. Aren't there strange things where the string "0" evaluates to false or something like that?

Comment (AndreasArvidsson) [reply packages/cursorless-engine/src/processTargets/modifiers/ItemStage/tokenizeRange.ts]: I don't believe so but I have no problem being more verbose if you prefer. Updated.

**Pull request merged**